packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -24,19 +26,24 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
 
       this.xhrTracker.onStart(this.trackRequest)
       this.fetchTracker.onStart(this.trackRequest)
+      this.networkRequestCallback = configuration.networkRequestCallback
     }
   }
 
   private trackRequest = (startContext: RequestStartContext): RequestEndCallback | undefined => {
     if (!this.shouldTrackRequest(startContext)) return
 
+    const networkRequestInfo = this.networkRequestCallback({ url: startContext.url, type: startContext.type })
+
+    if (!networkRequestInfo) return
+
     const span = this.spanFactory.startSpan(
       `[HTTP]/${startContext.method.toUpperCase()}`,
       { startTime: startContext.startTime, makeCurrentContext: false }
     )
 
     span.setAttribute('bugsnag.span.category', 'network')
-    span.setAttribute('http.url', startContext.url)
+    span.setAttribute('http.url', networkRequestInfo.url)
```

Comment:
What if someone returns a truthy value that doesn't have a `url` property? e.g. `true` or a random object